components/MW/ConvertedGeneID/index.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react'
+import { MetaNode } from '@/spec/metanode'
+import { z } from 'zod'
+import { VALID_LOADERS } from 'next/dist/shared/lib/image-config'
+
+export function uniqJsonSubset(data) {
+  let datastr = JSON.stringify(data);
+  datastr = datastr.replace("^[","");
+  datastr = datastr.replace("]$","");
+
+  const dataobj = JSON.parse(datastr);
+  let uniqENTREZID : string[] = Array.from(new Set( dataobj.map(a => a.ENTREZID)));
+  let uniqSYMBOL : string[] = Array.from(new Set( dataobj.map(a => a.SYMBOL)));
+  let uniqGENENAME : string[] = Array.from(new Set( dataobj.map(a => a.GENENAME)));
+
+  let uniqdataobj = uniqENTREZID.map((id, idx) => {
+   return {
+    ENTREZID : uniqENTREZID[idx], 
+    SYMBOL : uniqSYMBOL[idx],
+    GENENAME : uniqGENENAME[idx],
+  };
+  });
+  return uniqdataobj;
+}
+
+// A unique name for your data type is used here
+export const ConvertedGeneID = MetaNode.createData('ConvertedGeneID')
+  // Human readble descriptors about this node should go here
+  .meta({
+    label: 'Json for converted gene IDs table',
+    description: 'Json for converted gene IDs table',
+  })
+  // this should have a codec which can encode or decode the data type represented by this node
+  //  using zod, a compile-time and runtime type-safe codec can be constructed
+  .codec(z.any())
```

Comment:
I am not understanding this part. Where and how do I define the schema of the object (data?)?

Reply:
Here are some examples of other schemas:
https://github.com/nih-cfde/playbook-partnership/blob/main/components/service/mygeneinfo/index.tsx#L20-L29

https://github.com/nih-cfde/playbook-partnership/blob/main/components/core/significant_tissues/index.tsx#L5-L8

If you have an example of the data you can also use this:
https://rsinohara.github.io/json-to-zod-react/

To get you started -- it's basically the "shape" of your data.

Reply:
We will fix this in the next round; For now, I have made my edits directly into the files here on github, e.g. in
https://github.com/metabolomicsworkbench/playbook-partnership/edit/playbook-partnership-mano-20221129/components/MW/geneidconv/index.tsx

Sumana is doing the same.

Reply:
We have resolved all but three (all related to z.any), whee we need help. Can we do that in the next round after merging with nih-cfde:main, so that we can clone again and actually test locally while making the changes.

Reply:
@mano-at-sdsc this branch is up to date with main so you *can* test it locally.. you simply need to `git fetch && git pull` it into your local workspace.

Reply:
Thanks, @u8sand [keeping @susrinivasan in the loop]. I see so many branches on my local system. Can I start with a git clone. I got some warning/error in trying git fetch && git pull.

[mano@sc-cfdewebdev playbook-partnership]$ git fetch && git pull
remote: Enumerating objects: 90, done.
remote: Counting objects: 100% (48/48), done.
remote: Compressing objects: 100% (17/17), done.
remote: Total 90 (delta 38), reused 31 (delta 31), pack-reused 42
Unpacking objects: 100% (90/90), 18.51 KiB | 210.00 KiB/s, done.
From https://github.com/metabolomicsworkbench/playbook-partnership
   0e19dec..c6480bd  playbook-partnership-mano-20221129 -> origin/playbook-partnership-mano-20221129
 * [new branch]      main                               -> origin/main
Updating 0e19dec..c6480bd
error: Your local changes to the following files would be overwritten by merge:
        components/ConvertedGeneID/index.tsx
Please commit your changes or stash them before you merge.
Aborting
[mano@sc-cfdewebdev playbook-partnership]$

Reply:
>Please commit your changes or stash them before you merge.

```
git add --all .
git stash
```

Then you can pull. This is because your workspace has changes that would be overwritten if you were to pull.. If you care about your workspace changes afterwards you can run `git stash pop`

Reply:
Thanks, @u8sand. It worked and I see the UI. Giving some issue re reading json object, but I will debug that or ask you if I have trouble. @susrinivasan  will try the same. In her case, the local may not be pointing to playbook-partnership-mano-20221129 [ i.e., https://github.com/metabolomicsworkbench/playbook-partnership/tree/playbook-partnership-mano-20221129], so, does she need to specify from where to fetch/pull, as in

git fetch https://github.com/metabolomicsworkbench/playbook-partnership/tree/playbook-partnership-mano-20221129

Reply:
@mano-at-sdsc As long as she has the metabolomicsworkbench/playbook-partnership configured as a remote, she can check it out in a branch
```
git fetch remotename
git checkout -b somebranch -t remotename/somebranch
```

Reply:
@u8sand I added the schema and did
git add --all .
git stash

then

git fetch upstream
git pull

Now I don't see my latest changes in my file from today. How do I revert to the state before I did "git fetch upstream" and "git pull". If not possible, OK, I can just retype.

Reply:
No, @mano-at-sdsc, your changes are in the `stash` since you ran `git stash` you can get them back with `git stash pop`

Reply:
Thanks @u8sand . I got back my edits from today. Now, I want to bring in any recent changes from nih-cfde/playbook-partnership to my local copy in the current branch.
C[mano@sc-cfdewebdev playbook-partnership]$ git remote -v
mano-at-sdsc/playbook-partnership       https://github.com/mano-at-sdsc/playbook-partnership (fetch)
mano-at-sdsc/playbook-partnership       https://github.com/mano-at-sdsc/playbook-partnership (push)
origin  https://github.com/metabolomicsworkbench/playbook-partnership.git (fetch)
origin  https://github.com/metabolomicsworkbench/playbook-partnership.git (push)
upstream        https://github.com/nih-cfde/playbook-partnership/ (fetch)
upstream        https://github.com/nih-cfde/playbook-partnership/ (push)

I suppose origin actually refers to
https://github.com/metabolomicsworkbench/playbook-partnership/tree/playbook-partnership-mano-20221129
even though it just says https://github.com/metabolomicsworkbench/playbook-partnership.git since the local active branch is playbook-partnership-mano-20221129.

I have taken a manual backup of my files. I suppose I should do:

git add --all .
git stash

git fetch upstream
git merge upstream/main

Please let me know. Thanks.

Reply:
I agree. I will sync with Sumana as well and then schedule something to make sure we are on the right track. Just now, following what we did last time, I pushed local changes to origin/playbook-partnership-mano-20221129 and I see it on github: https://github.com/metabolomicsworkbench/playbook-partnership/tree/playbook-partnership-mano-20221129. Then, when I did git fetch upstream && git merge upstream, it seems to have incorporated some changes related to glygen from nih-cfde/playbook-partnership into my local copy [looks like I didn't have gly_gen folder in my local copy earlier]. So far, it seems to have gone as expected. Thanks.

Reply:
Hi @u8sand . We have pushed the updated metgene files as well (schema added and codec argument set correctly) to  https://github.com/metabolomicsworkbench/playbook-partnership/tree/playbook-partnership-mano-20221129. Would you like to meet over zoom sometime, may be next week, where we can walk over the recent changes, or you can review it yourself. I think these are now ready to be integrated into your master repo. Thanks.